File: Java/977. 有序数组的平方.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int l = 0;
        int r = nums.length - 1;
        int[] res = new int[nums.length];
        int k = r;
        while(l <= r) {
            if (nums[l] * nums[l] >= nums[r] * nums[r]) {
                res[k] = nums[l] * nums[l];
                l++;
            } else {
                res[k] = nums[r] * nums[r];
                r--;
            }
            k--;
        }
        return res;
    }
}

File: python/977. 有序数组的平方.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        l, r = 0, len(nums)-1
        res = [0]*len(nums)
        k = r
        while l <= r:
            if nums[l] * nums[l] >= nums[r] * nums[r]:
                res[k] = nums[l] * nums[l]
                l += 1
            else:
                res[k] = nums[r] * nums[r]
                r -= 1
            k -= 1
        return res
